src/babashka/pods/sci.clj

```diff
@@ -19,6 +24,39 @@
               (string? var-value)
               (sci/eval-string* ctx var-value))))))
 
+(defn metadata-cache-file ^File [pod-spec {:keys [:version]}]
+  (io/file (resolver/cache-dir {:pod/name pod-spec :pod/version version})
+           "metadata.cache"))
+
+(defn load-metadata-from-cache [pod-spec opts]
+  (let [cache-file (metadata-cache-file pod-spec opts)]
+    (when (.exists cache-file)
+      (with-open [r (PushbackInputStream. (io/input-stream cache-file))]
+        (impl/read r)))))
+
+(defn load-pod-metadata* [pod-spec {:keys [:version] :as opts}]
+  (let [metadata (impl/load-pod-metadata pod-spec opts)
+        cache-file (when (qualified-symbol? pod-spec) ; don't cache local pods b/c their namespaces can change
+                     (metadata-cache-file pod-spec opts))]
+    (when cache-file
+      (with-open [w (io/output-stream cache-file)]
+        (impl/write w metadata)))
+    metadata))
+
+(defn load-pod-metadata [pod-spec {:keys [:version] :as opts}]
+  (let [metadata
+        (if-let [cached-metadata (when (qualified-symbol? pod-spec) ; don't cache local pods b/c their namespaces can change
+                                   (load-metadata-from-cache pod-spec opts))]
+          cached-metadata
+          (load-pod-metadata* pod-spec opts))]
+    (reduce
+      (fn [pod-namespaces ns]
+        (let [ns-sym (-> ns (get "name") impl/bytes->string symbol)]
+          (assoc pod-namespaces ns-sym {:pod-spec pod-spec
+                                        :opts (assoc opts :metadata metadata)})))
```

Comment:
should we leave metadata out of opts and just use its own key for it? seems a little cleaner since those opts are going to be passed to pods and pods aren't really interested in their own metadata.

Reply:
Sure, that seems fine. But it's a little tricky how to pass it to `b.pods.sci/load-pod` then b/c that already has a 4-arity arg vector where version is its own arg.

I could dissoc metadata from opts after we use it in `b.pods.impl/load-pod`.

Reply:
@cap10morgan Not sure if I understand. The format I was suggesting:

```
{:pod-spec ...
 :opts ...
 :metadata ...}
```

How does that relate to a call to `load-pod`?

Reply:
Because ultimately we have to get that metadata into `b.pods.impl/load-pod` (specifically here: https://github.com/cap10morgan/pods/blob/c2a2facb0fa2d4139445929de8d42c149e10c88d/src/babashka/pods/impl.clj#L394) or it will just redundantly call `:describe` again.

But to get it there, it first has to pass through `b.pods.sci/load-pod` which has the following arities defined: https://github.com/babashka/pods/blob/538fc6f4147badf3a457b3eb8f9f95dd3bc947b1/src/babashka/pods/sci.clj#L22-L25

I might be missing something simple, but to pass something else to `b.pods.sci/load-pod` it seems like we either need it to be in `opts` or we need add an additional arg to (at least) the 2nd and 3rd arg vectors so they become 5-arity and 4-arity. Otherwise the 4-arity would get ambiguous between `version` and `metadata` args.

Reply:
>  it will just redundantly call :describe again.

I wasn't aware that you were actually bypassing the describe call and using the cached file. I think it doesn't matter that it will call describe again, it will probably not be any slower than reading the file from disk. 

I'd rather have this PR not change too much. All we really need is to know the namespaces the pod exposes so we can load the pod on require, nothing more than that. To accomplish this we only need the following changes:

1) Cache the namespaces in a file (namespaces.edn or whatever). We decided to cache the whole response instead, which wasn't really necessary, but this is an implementation detail.

Reply:
Let's leave it like you have it now though.

Reply:
Never mind what I said earlier, I think it's fine. I was just worried that a user might override this option since options is merged with whatever the user provides.